src/commands/context/useDockerContext.ts

```diff
@@ -5,15 +5,27 @@
 
 import * as vscode from 'vscode';
 import { IActionContext } from 'vscode-azureextensionui';
+import { ext } from '../../extensionVariables';
 import { localize } from '../../localize';
+import { ContextTreeItem } from '../../tree/contexts/ContextTreeItem';
 import { dockerContextManager } from '../../utils/dockerContextManager';
-import { selectDockerContext } from './selectDockerContext';
 
-export async function useDockerContext(_actionContext: IActionContext): Promise<void> {
-    const selectedContext = await selectDockerContext(localize('vscode-docker.commands.context.selectContextToUse', 'Select Docker context to inspect'));
+export async function useDockerContext(actionContext: IActionContext, node?: ContextTreeItem): Promise<void> {
+    let invokedFromCommandPalette = false;
+    if (!node) {
+        node = await ext.contextsTree.showTreeItemPicker<ContextTreeItem>(ContextTreeItem.contextValue, {
+            ...actionContext,
+            noItemFoundErrorMessage: localize('vscode-docker.commands.contexts.use.noContexts', 'No contexts are available to use')
+        });
+        invokedFromCommandPalette = true;
+    }
 
-    await dockerContextManager.use(selectedContext.Name);
+    await node.use(actionContext);
+    await ext.contextsRoot.refresh();
+    dockerContextManager.expediteContextCheck();
```

Comment:
Call to `dockerContextManager.use()` does `expediteContextCheck()` internally, so no need to call it explicitly here.